app/models/library_entry.rb

```diff
@@ -241,54 +239,36 @@ def kind
   end
 
   after_create do
-    # Activity History Stat will be using this
-    library_event = LibraryEvent.create_for(:added, self)
-    # Stat STI
-    case kind
-    when :anime
-      Stat::AnimeCategoryBreakdown.increment(user, self)
-      Stat::AnimeAmountConsumed.increment(user, self)
-      Stat::AnimeFavoriteYear.increment(user, self)
-    when :manga
-      Stat::MangaCategoryBreakdown.increment(user, self)
-      Stat::MangaAmountConsumed.increment(user, self)
-      Stat::MangaFavoriteYear.increment(user, self)
-    end
+    Stat::LibraryCreateWorker.perform_async(
+      kind, user, id,
+      options: {
+        progress: progress,
+        progress_was: progress_was,
+        progress_changed: progress_changed?
+      }
+    )
   end
 
   after_update do
-    # Activity History Stat will be using this
-    library_event = LibraryEvent.create_for(:updated, self)
-    case kind
-    when :anime
-      # special case checking if progress was increased or decreased
-      if progress > progress_was
-        Stat::AnimeAmountConsumed.increment(user, self, true)
-      elsif progress < progress_was
-        Stat::AnimeAmountConsumed.decrement(user, self, true)
-      end
-    when :manga
-      # special case checking if progress was increased or decreased
-      if progress > progress_was
-        Stat::MangaAmountConsumed.increment(user, self, true)
-      elsif progress < progress_was
-        Stat::MangaAmountConsumed.decrement(user, self, true)
-      end
-    end
+    Stat::LibraryUpdateWorker.perform_async(
+      kind, user, id,
+      options: {
+        progress: progress,
+        progress_was: progress_was,
+        progress_changed: progress_changed?
+      }
+    )
   end
 
   after_destroy do
-    # Stat STI
-    case kind
-    when :anime
-      Stat::AnimeCategoryBreakdown.decrement(user, self)
-      Stat::AnimeAmountConsumed.decrement(user, self)
-      Stat::AnimeFavoriteYear.decrement(user, self)
-    when :manga
-      Stat::MangaCategoryBreakdown.decrement(user, self)
-      Stat::MangaAmountConsumed.decrement(user, self)
-      Stat::MangaFavoriteYear.decrement(user, self)
-    end
+    Stat::LibraryDestroyWorker.perform_async(
+      kind, user, self,
```

Comment:
Can't do `self`, need to pass `id` instead

Reply:
uhhh... the id won't exist because it would be deleted.

Reply:
Then neither will `self`